stdlib/http/src/main/ballerina/src/http/client_endpoint.bal

```diff
@@ -472,27 +494,48 @@ function createRetryClient(string url, ClientConfiguration configuration) return
             maxWaitIntervalInMillis: retryConfig.maxWaitIntervalInMillis,
             statusCodes: statusCodes
         };
-        if (configuration.cache.enabled) {
-            var httpCachingClient = createHttpCachingClient(url, configuration, configuration.cache);
-            if (httpCachingClient is HttpClient) {
-                return new RetryClient(url, configuration, retryInferredConfig, httpCachingClient);
-            } else {
-                return httpCachingClient;
-            }
+        var httpCookieClient = createCookieClient(url, configuration, cookieStore);
+        if (httpCookieClient is HttpClient) {
+            return new RetryClient(url, configuration, retryInferredConfig, httpCookieClient);
         } else {
-            var httpSecureClient = createHttpSecureClient(url, configuration);
-            if (httpSecureClient is HttpClient) {
-                return new RetryClient(url, configuration, retryInferredConfig, httpSecureClient);
-            } else {
-                return httpSecureClient;
-            }
+            return httpCookieClient;
         }
     } else {
-        //remove following once we can ignore
-        if (configuration.cache.enabled) {
-            return createHttpCachingClient(url, configuration, configuration.cache);
+        return createCookieClient(url, configuration, cookieStore);
+    }
+}
+
+function createCookieClient(string url, ClientConfiguration configuration, CookieStore cookieStore) returns HttpClient|ClientError {
+    var cookieConfigVal = configuration.cookieConfig;
+    if (cookieConfigVal is CookieConfig) {
```

Comment:
shall we follow the fail fast pattern? so you can simplify the logic to have less "else" blocks